assets/admin/js/forms.js

```diff
@@ -1,5 +1,122 @@
 jQuery( document ).ready( function($) {
 
+	var jb_media_uploader;
+	$( document.body ).on( 'click', '.jb-change-media', function(e) {
+		e.preventDefault();
+		$(this).parents('.jb-uploaded-wrapper').removeClass('jb-uploaded').addClass('jb-waiting-change');
+		$(this).parents('.jb-uploaded-wrapper').siblings('.jb-uploader').removeClass('jb-uploaded');
+	});
+
+	$( document.body ).on( 'click', '.jb-cancel-change-media', function(e) {
+		e.preventDefault();
+		$(this).parents('.jb-uploaded-wrapper').addClass('jb-uploaded').removeClass('jb-waiting-change');
+		$(this).parents('.jb-uploaded-wrapper').siblings('.jb-uploader').addClass('jb-uploaded');
+	});
+
+	$( document.body ).on( 'click', '.jb-clear-media', function(e) {
+		e.preventDefault();
+		jb_media_uploader.splice();
+		$(this).parents('.jb-uploaded-wrapper').siblings('.jb-media-value').val('');
+		$(this).parents('.jb-uploaded-wrapper').siblings('.jb-media-value-hash').val('');
+		$(this).parents('.jb-uploaded-wrapper').removeClass('jb-uploaded').removeClass('jb-waiting-change');
+		$(this).parents('.jb-uploaded-wrapper').siblings('.jb-uploader').removeClass('jb-uploaded');
+	});
+
+	$('.jb-select-media').each( function() {
+		var $button = $(this);
+		var $action = $button.data('action');
+		var $filelist = $button.parents('.jb-uploader-dropzone');
+		var $button_wrapper = $button.parents('.jb-select-media-button-wrapper');
+		var $errorlist = $filelist.siblings( '.jb-uploader-errorlist' );
+		var extensions = 'jpg,jpeg,gif,png,bmp,ico,tiff';
+
+		var uploader_args = {
+			browse_button: $button.get( 0 ), // you can pass in id...
+			drop_element: $filelist.get( 0 ), // ... or DOM Element itself
+			container: $button_wrapper.get( 0 ), // ... or DOM Element itself
+			url: wp.ajax.settings.url + '?action=' + $action + '&nonce=' + jb_front_data.nonce,
+			chunk_size: '1024kb',
+			max_retries: 1,
+			multipart: true,
+			multi_selection: false,
+			filters: {
+				max_file_size: '10mb',
+				mime_types: [
+					{ title: wp.i18n.__( 'Image files', 'jobboardwp' ), extensions: extensions },
```

Comment:
Based on line 31 there is possible to upload only images. What about if the field is set up to upload not-image files?
It should be dynamic from data attribute